### ISSUE TITLE
Add option to apply WS_BORDER style for frameless window

### ISSUE THREAD

**Issue by bsclifton** [electron/electron#7172]
- Electron version:
  latest
- Operating system:
  Windows (7 and 10)

The current BrowserWindow implementation doesn't have a way to optionally add back in the borders.  In Windows API, I'm familiar with having these by using styles like:
WS_BORDER, WS_DLGFRAME

and/or extended styles:
WS_EX_DLGMODALFRAME | WS_EX_CLIENTEDGE | WS_EX_STATICEDGE

Windows 10 puts a 1 pixel border around the entire window with the user's selected color. When the app is missing this, it doesn't feel like a native app.

Here's an extreme close-up showing a frameless electron window (back) and a native windows app (front)
![image](https://cloud.githubusercontent.com/assets/4733304/18412295/8ff55a22-773f-11e6-8dea-90e4e02ffa7d.png)

**Comment by bsclifton** [comment]
I've tried adding the styles manually here:
https://github.com/electron/electron/blob/master/atom/browser/native_window_views.cc#L283

But I think they're not being taken into consideration because I'm thinking the non-client area is being eliminated completely (by handling the approprate NC_ events)

**Comment by bbondy** [comment]
You can't draw them via front end?

**Comment by bsclifton** [comment]
Can definitely be done in the front end with CSS :) But long term, would be cool to have this available so that our app can have the exact styles from the users theme

**Comment by bbondy** [comment]
ok if this isn't straight forward though we should just do in CSS since we already do menu that way and window buttons.

**Comment by bsclifton** [comment]
Yup, my work-around is already in progress- just wanted to open this up in case folks wanted to comment or if anyone was wanting to grab this task

**Comment by fuermosi777** [comment]
@bsclifton Could you please share how did you manually add the border? Thanks.

**Comment by bsclifton** [comment]
@fuermosi777 I never solved the underlying issue in Electron. Instead, I worked around by using CSS to apply the border and then capturing the window blur to turn borders gray when not focused

**Comment by fuermosi777** [comment]
@bsclifton Thanks. I tried the CSS way but looks like Electron on Windows 10 has a bug that causes right and bottom 1px border disappear when user is under "DPI zooming mode" (e.g 125%). https://github.com/electron/electron/issues/10862

**Comment by binaryfunt** [comment]
Any chance this can be looked at? The bug mentioned by @fuermosi777 (actually a Chromium bug - I can reproduce it in Google Chrome 80) means the CSS workaround doesn't work very well (not to mention media queries are needed for higher scaling factors to keep the border to 1px, as well as the fact that the native Windows 10 border now has a degree of transparency for underlying content that is impossible to reproduce). Not having a border means the content can blend into the background, which is bad for usability.
